yt/data_objects/data_containers.py

```diff
@@ -1441,6 +1441,8 @@ def _tupleize_field(self, field):
         raise YTFieldNotParseable(field)
 
     def _determine_fields(self, fields):
+        if str(fields) in self.ds._determined_fields:
+            return self.ds._determined_fields[str(fields)]
         explicit_fields = []
```

Comment:
Any reason not to use the `functools.cache` here?

Reply:
The trick is the `self` part changes a lot and I'm not sure how to do that with `functools.cache` but pointers are welcome.

Reply:
quick note: `functools.cache` is Python >= 3.9
For now we need to stay with `functools.lru_cache`

Reply:
Setting this aside, I'm confused why you're seemingly caching groups of fields as keys instead of individual fields ?

Reply:
`fields` can sometimes be a tuple, sometimes be a list with the same single tuple, sometimes be a list of the same single tuple and something else. It's OK to miss cache. It's not OK to add complicated logic that will determine whether I've seen that tuple before or not in some weird combination of python objects, if I'm looking for a performance gain.

Reply:
alright, sounds good :)